src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiOptionsFormField.js

```diff
@@ -0,0 +1,360 @@
+    </CustomFieldTemplate>
+  </form>
+);
+
+const ObjectsApiFormFields = ({index, name, schema, formData, onChange}) => {
+  const intl = useIntl();
+  const validationErrors = useContext(ValidationErrorContext);
+
+  const [availableObjecttypes, setAvailableObjecttypes] = useState([]);
+  const [availableObjecttypeVersions, setAvailableObjecttypeVersions] = useState([]);
+
+  const {
+    objecttype = '',
+    objecttypeVersion = '',
+    productaanvraagType = '',
+    informatieobjecttypeSubmissionReport = '',
+    uploadSubmissionCsv = '',
+    informatieobjecttypeSubmissionCsv = '',
+    informatieobjecttypeAttachment = '',
+    organisatieRsin = '',
+    contentJson = '',
+    paymentStatusUpdateJson = '',
+  } = formData;
+
+  // For existing registration options where the URL might not be valid
+  // will default to a textinput in that case
+  const unknownObjecttype = useMemo(
+    () => objecttype && !availableObjecttypes.map(o => o.url).includes(objecttype),
+    [availableObjecttypes, objecttype]
+  );
+
+  const onFieldChange = event => {
+    const {name, value} = event.target;
+    const updatedFormData = produce(formData, draft => {
+      draft[name] = value;
+    });
+    onChange(updatedFormData);
+  };
+
+  const buildErrorsComponent = field => {
+    const rawErrors = getFieldErrors(name, index, validationErrors, field);
+    return rawErrors ? getErrorMarkup(rawErrors) : null;
+  };
+
+  // API fetches:
+  const {loading: objecttypesLoading} = useAsync(async () => {
+    const response = await get(REGISTRATION_OBJECTTYPES_ENDPOINT);
+    if (!response.ok) {
+      throw new Error('Error when loading objecttypes');
+    }
+
+    setAvailableObjecttypes(response.data);
+  });
+
+  const {loading: objecttypeVersionLoading} = useAsync(async () => {
+    if (!objecttype) return;
+
+    const response = await get(
+      REGISTRATION_OBJECTTYPE_VERSIONS_ENDPOINT.replace(
+        '{objecttypeUuid}',
+        availableObjecttypes.find(o => o.url === objecttype).uuid
+      )
+    );
+    if (!response.ok) {
+      throw new Error('Error when loading objecttype versions');
+    }
+
+    setAvailableObjecttypeVersions(response.data);
+  }, [objecttype]);
+
+  return (
+    <Wrapper>
+      <CustomFieldTemplate
+        id="root_objecttype"
+        label={intl.formatMessage({
+          defaultMessage: 'Objecttype',
+          description: 'Objects API registration options "Objecttype" label',
+        })}
+        rawDescription={intl.formatMessage({
+          defaultMessage:
+            'URL that points to the ProductAanvraag objecttype in the Objecttypes API. The objecttype should have the following three attributes: "submission_id", "type" (the type of productaanvraag) and "data" (the submitted form data)',
+          description: 'Objects API registration options "Objecttype" description',
+        })}
+        rawErrors={getFieldErrors(name, index, validationErrors, 'objecttype')}
```

Comment:
This `rawErrors` prop is weird: it is described as an array of string, but is only used in a "truthy" context (`if rawErrors`/`rawErrors && ...`)

Reply:
yep, we're stuck with it until we migrate to Formik unfortunately